js/interface.js

```diff
@@ -536,9 +536,27 @@ $('#app')
   })
   .on('click', '[data-delete-source]', function(event) {
     event.preventDefault();
-    var confirmAlert = confirm('Are you sure you want to delete this data source? All entries will be deleted.');
+    var message = 'Are you sure you want to delete this data source? All entries will be deleted.';
+    var currentDS = _.find(dataSources, function(ds) {
+      return ds.id === currentDataSourceId;
+    });
+
+    if (currentDS.apps.length) {
+      var appPrefix = currentDS.apps.length > 1 ? 'apps: ' : 'app: ';
+      var appUsedIn = currentDS.apps.map(function(elem) {
+        return ' ' + elem.name;
+      });
+
+      message = 'Are you sure you want to delete this data source? The data source is used in the next ' + appPrefix + appUsedIn + '. All entries will be deleted.';
```

Comment:
"The data source is currently in use by the following apps: A, B, C."

Reply:
Fixed